src/cascadia/TerminalSettingsEditor/NullableColorPicker.cpp

```diff
@@ -0,0 +1,233 @@
+        // Initialize any dependency properties here.
+        // This performs a lazy load on these properties, instead of
+        // initializing them when the DLL loads.
+        if (!_ColorSchemeVMProperty)
+        {
+            _ColorSchemeVMProperty =
+                DependencyProperty::Register(
+                    L"ColorSchemeVM",
+                    xaml_typename<Editor::ColorSchemeViewModel>(),
+                    xaml_typename<Editor::NullableColorPicker>(),
+                    PropertyMetadata{ nullptr });
+        }
+        if (!_CurrentColorProperty)
+        {
+            _CurrentColorProperty =
+                DependencyProperty::Register(
+                    L"CurrentColor",
+                    xaml_typename<Windows::Foundation::IReference<Microsoft::Terminal::Core::Color>>(),
+                    xaml_typename<Editor::NullableColorPicker>(),
+                    PropertyMetadata{ nullptr, PropertyChangedCallback{ &NullableColorPicker::_OnCurrentColorValueChanged } });
+        }
+        if (!_ShowNullColorButtonProperty)
+        {
+            _ShowNullColorButtonProperty =
+                DependencyProperty::Register(
+                    L"ShowNullColorButton",
+                    xaml_typename<bool>(),
+                    xaml_typename<Editor::NullableColorPicker>(),
+                    PropertyMetadata{ box_value(true) });
+        }
+        if (!_NullColorButtonLabelProperty)
+        {
+            _NullColorButtonLabelProperty =
+                DependencyProperty::Register(
+                    L"NullColorButtonLabel",
+                    xaml_typename<hstring>(),
+                    xaml_typename<Editor::NullableColorPicker>(),
+                    PropertyMetadata{ nullptr });
+        }
+        if (!_NullColorPreviewProperty)
+        {
+            _NullColorPreviewProperty =
+                DependencyProperty::Register(
+                    L"NullColorPreview",
+                    xaml_typename<Windows::UI::Color>(),
+                    xaml_typename<Editor::NullableColorPicker>(),
+                    PropertyMetadata{ box_value(Windows::UI::Colors::Transparent()) });
+        }
+    }
+
+    void NullableColorPicker::_OnCurrentColorValueChanged(const DependencyObject& d, const DependencyPropertyChangedEventArgs& /*e*/)
+    {
+        const auto& obj{ d.try_as<Editor::NullableColorPicker>() };
+        get_self<NullableColorPicker>(obj)->_UpdateColorChips();
+    }
+
+    void NullableColorPicker::_UpdateColorChips()
+    {
+        const auto& currentColor = CurrentColor();
+        for (const auto& colorChip : _colorChips)
+        {
+            const auto& chipColor = colorChip.DataContext().as<Editor::ColorTableEntry>().Color();
+            colorChip.IsChecked(currentColor ?
+                                    chipColor.R == currentColor.Value().R && chipColor.G == currentColor.Value().G && chipColor.B == currentColor.Value().B :
+                                    false);
```

Comment:
Does `==` not work? If it doesn't, you can also do a good old `memcmp(...) == 0` if you'd like. 😅

Reply:
Nah, we're comparing a `Windows::UI::Color` to a `Microsoft::Terminal::Core::Color`